spm_results_export.m

```diff
@@ -15,26 +16,54 @@
     TabDat = spm_list('Table',xSPM);
 end
 
-%If we're using matlab SPM, xSPM and TabDat objects we need to generate the 
-%MIP and design matrix manually.
-if ~isfield(xSPM, 'nidm')
-    MIP     = spm_mip(xSPM.Z,xSPM.XYZmm,xSPM.M,xSPM.units);
-end
-if ~isfield(SPM, 'nidm')
-    DesMtx  = (SPM.xX.nKX + 1)*32;
-end
-
 %If we're using the matlab made SPM, xSPM and TabDat objectes just output
 %into the current directory, else output next to the NIDM objects.
 
 if ~isfield(SPM, 'nidm')
-    outdir  = spm_file(fullfile(pwd, 'temp'), 'unique');
+    fHTML = pwd;
+    outdir  = spm_file(fullfile(pwd, 'temp'));
 else
-    outdir  = spm_file(fullfile(SPM.nidm.filepath,'temp'), 'unique');
+    fHTML = SPM.nidm.filepath;
+    outdir  = spm_file(fullfile(SPM.nidm.filepath,'temp'));
+end
+ 
+if exist(outdir) ~= 7
+    outdir = spm_file(outdir, 'uniquedir');
+end
+
+fHTML   = spm_file(fullfile(fHTML, 'index.html'));
+
+%If fHTML already exists, ask if it should be overWritten.
+
+overWrite = true;
+if exist(fHTML, 'file') == 2
+    button = questdlg('The output file, index.html, already exists. Would you like this file to be overwritten?', 'Warning', 'Overwrite', 'Do not overwrite', 'Do not overwrite');      
+    switch button
+        case 'Overwrite'
+            overWrite = true;
+        case 'Do not overwrite'
+            overWrite = false;
+        case ''
+            overWrite = false;
+    end
+end
+
+if(~overWrite)

```

Comment:
This can go inside the if above so that my last comment (setting `overWrite` to false by default) does not cause issue here.